### PR TITLE
[WIP] Check for empty message in Envelope class

### PR DESCRIPTION
**Pull request by alexvaluyskiy**

### CODE REVIEW

File: src/core/Akka/Actor/ActorCell.cs [hunk 1]
```diff
@@ -308,33 +308,31 @@ public virtual void SendMessage(IActorRef sender, object message)
...
-
+                    Serializer serializer = _systemImpl.Serialization.FindSerializerFor(message.Message);
+                    byte[] serialized = serializer.ToBinary(message.Message);
 
                     var manifestSerializer = serializer as SerializerWithStringManifest;
                     if (manifestSerializer != null)
                     {
                         var manifest = manifestSerializer.Manifest(serialized);
-                        message = _systemImpl.Serialization.Deserialize(serialized, manifestSerializer.Identifier, manifest);
+                        message = new Envelope(_systemImpl.Serialization.Deserialize(serialized, manifestSerializer.Identifier, manifest), message.Sender);
```
Comment (alexvaluyskiy) [inline src/core/Akka/Actor/ActorCell.cs]: Scala version has a bit different implementation
https://github.com/akka/akka/blob/4acc1cca6a27be0ff80f801de3640f91343dce94/akka-actor/src/main/scala/akka/actor/dungeon/Dispatch.scala#L138

I should check for null, and then throw an exception

File: src/core/Akka.FSharp.Tests/ApiTests.fs [hunk 2]
```diff
@@ -121,8 +121,8 @@ let ``actor that accepts _ will receive string message`` () =
     response
     |> equals "SomethingToReturn"
 
-[<Fact>]
-// SUCCEEDS
+//[<Fact>]
+// FAILS
 let ``actor that accepts unit will receive unit message`` () =
```
Comment (alexvaluyskiy) [inline src/core/Akka.FSharp.Tests/ApiTests.fs]: I've discussed about this test with @Horusiath . He told me, that I should remove this test if it is not actual at the moment

File: src/core/Akka/Actor/Message.cs [hunk 3]
```diff
@@ -5,26 +5,43 @@
...
+            if (message == null)
+            {
+                throw new ArgumentNullException(nameof(message));
+            }
+
+            Message = message;
+            Sender = sender != ActorRefs.NoSender ? sender : system.DeadLetters;
+        }
+
+        public Envelope(object message, IActorRef sender)
```
Comment (alexvaluyskiy) [inline src/core/Akka/Actor/Message.cs]: I should have made this constructor as internal, but I did not manage to do it. I've got some conflicts in Persistence

**Comment by Aaronontheweb** [comment]
Why was this merged @alexvaluyskiy ? Did someone else review it?

**Comment by alexvaluyskiy** [comment]
I've reverted it back

**Comment by alexvaluyskiy** [comment]
@Aaronontheweb look at this https://github.com/akkadotnet/akka.net/pull/2210

**Pull request merged**